psi4/src/psi4/psimrcc/mrccsd_t_form_matrices.cc

```diff
@@ -43,43 +43,40 @@
 #include "mrccsd_t.h"
 #include "special_matrices.h"
 
-extern FILE* outfile;
-
 namespace psi {
 namespace psimrcc {
-extern MOInfo* moinfo;
 
 // Creates the matrix T2[ij][a][b] with irreps ordered according to b
 void MRCCSD_T::form_T2_ij_a_b(IndexMatrix* T2_ij_a_b, bool spin1, bool spin2, bool transpose) {
-    CCIndexIterator ij("[oo]");
+    CCIndexIterator ij(wfn_, "[oo]");
 
     // Copy the matrix elements
     for (int ref = 0; ref < nrefs; ++ref) {
-        int unique_ref = moinfo->get_ref_number(ref, AllRefs);
+        int unique_ref = wfn_->moinfo()->get_ref_number(ref, AllRefs);
 
         std::vector<double**> Tijab;
 
         if (unique_ref == ref) {
             if (spin1 && spin2) {
-                Tijab = blas->get_MatTmp("t2[oo][vv]", ref, none)->get_matrix();
+                Tijab = wfn_->blas()->get_MatTmp("t2[oo][vv]", ref, none)->get_matrix();
             } else if (!spin1 && !spin2) {
-                Tijab = blas->get_MatTmp("t2[OO][VV]", ref, none)->get_matrix();
+                Tijab = wfn_->blas()->get_MatTmp("t2[OO][VV]", ref, none)->get_matrix();
             } else if (spin1 && !spin2) {
-                Tijab = blas->get_MatTmp("t2[oO][vV]", ref, none)->get_matrix();
+                Tijab = wfn_->blas()->get_MatTmp("t2[oO][vV]", ref, none)->get_matrix();
             }
         } else {
             if (spin1 && spin2) {
-                Tijab = blas->get_MatTmp("t2[OO][VV]", unique_ref, none)->get_matrix();
+                Tijab = wfn_->blas()->get_MatTmp("t2[OO][VV]", unique_ref, none)->get_matrix();
             } else if (!spin1 && !spin2) {
-                Tijab = blas->get_MatTmp("t2[oo][vv]", unique_ref, none)->get_matrix();
+                Tijab = wfn_->blas()->get_MatTmp("t2[oo][vv]", unique_ref, none)->get_matrix();
             } else if (spin1 && !spin2) {
-                Tijab = blas->get_MatTmp("t2[Oo][Vv]", unique_ref, none)->get_matrix();
+                Tijab = wfn_->blas()->get_MatTmp("t2[Oo][Vv]", unique_ref, none)->get_matrix();
             }
         }
 
         for (ij.first(); !ij.end(); ij.next()) {
-            BlockMatrix* block_matrix = new BlockMatrix(nirreps, v->get_tuplespi(), v->get_tuplespi(), ij.sym());
-            CCIndexIterator ab("[vv]", ij.sym());
+            BlockMatrix* block_matrix = new BlockMatrix(wfn_, v->get_tuplespi(), v->get_tuplespi(), ij.sym());
```

Comment:
At present, yes. `BlockMatrix` needs to be able to create `CCIndexIterator` objects. All four constructors need `nirrep`, but two of the four also need a `CCBLAS` object. To me, the easiest solution was to have all four of the `CCIndexIterator` constructors take in a `wfn` object, which means `BlockMatrix` gets a wavefunction. I can think of a couple ways to avoid passing a `wfn` to BlockMatrix, but the current solution doesn't seem to be a problem to me.

Same response to below.

Reply:
Nope, if it's not a typo, it's not a problem.